packages/eslint-plugin/src/rules/object-curly-spacing.ts

```diff
@@ -20,6 +20,7 @@ export type MessageIds = InferMessageIdsTypeFromRule<typeof baseRule>;
 
 export default createRule<Options, MessageIds>({
   name: 'object-curly-spacing',
+  // eslint-disable-next-line eslint-plugin/prefer-message-ids,eslint-plugin/require-meta-type,eslint-plugin/require-meta-schema,eslint-plugin/require-meta-fixable -- all in base rule - https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/issues/274
```

Comment:
Hmm, can these rules be made type-aware? Would that fix their understanding of base rule extends?

Reply:
Good question. I think a number of improvements could be made here:

1. As a quick fix, I'm planning to update these rules to ignore meta spreads that we can't easily evaluate.
   * https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/issues/274 
2. eslint-plugin-eslint-plugin currently checks to ensure rules have the proper meta properties with acceptable values. I believe TypeScript's type-checking of the call to the @typescript-eslint/eslint-plugin `createRule()` helper enforces some of the same constraints, so we could possibly defer to TypeScript for some of this or utilize the type information in our checks.
3. But eslint-plugin-eslint-plugin also tries to find the values of various properties (for example, checking to see if `meta.fixable` is enabled). The type information alone wouldn't be enough to tell us what the actual value is. Perhaps there are TypeScript static analysis utilities that could help us determine the value in some situations where we can't today, but for complex/dynamic code, we can't always determine what values will be used.

Reply:
Cool, thanks for all the info! Same request as other places - if there's a rule disable, it should have a link to an issue or similar tracking making work to make the disable no longer necessary.

Reply:
Yep, already added a link to all disable comments.